service/src/main/java/org/kiwiproject/champagne/resource/HostConfigurationResource.java

```diff
@@ -0,0 +1,80 @@
+package org.kiwiproject.champagne.resource;
+
+import java.util.List;
+
+import javax.annotation.security.PermitAll;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.lang3.StringUtils;
+import org.kiwiproject.champagne.dao.AuditRecordDao;
+import org.kiwiproject.champagne.dao.HostDao;
+import org.kiwiproject.champagne.model.Host;
+import org.kiwiproject.champagne.model.AuditRecord.Action;
+
+import com.codahale.metrics.annotation.ExceptionMetered;
+import com.codahale.metrics.annotation.Timed;
+
+@Path("/host")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+@PermitAll
+public class HostConfigurationResource extends AuditableResource {
+
+    private final HostDao hostDao;
+
+    public HostConfigurationResource(HostDao hostDao, AuditRecordDao auditRecordDao) {
+        super(auditRecordDao);
+
+        this.hostDao = hostDao;
+    }
+    
+    @GET
+    @Path("/{environment}")
+    @Timed
+    @ExceptionMetered
+    public Response listHostsForEnvironment(@PathParam("environment") Long envId, @QueryParam("componentFilter") String componentFilter) {
+        List<Host> hosts;
+
+        if (StringUtils.isBlank(componentFilter)) {
```

Comment:
Consider static import and even using a ternary?

```suggestion
var hosts = isBlank(componentFilter) ? List.of() : hostDao.findHostsByEnvId(envId);
```

I am assuming the compiler will be able to infer the type here...